SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoUpdateSettings(FString DevAuthToken)
+{
+	AsyncTask(ENamedThreads::GameThread, [this, DevAuthToken]()
+		{
+			USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+			GDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
+			GDKEditorSettings->SaveConfig();
+			GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
+
+			// Ensure we enable bUseDevelopmentAuthenticationFlow when using cloud deployment flow.
+			USpatialGDKSettings* GDKRuntimeSettings = GetMutableDefault<USpatialGDKSettings>();
+			GDKRuntimeSettings->bUseDevelopmentAuthenticationFlow = true;
+			GDKRuntimeSettings->DevelopmentAuthenticationToken = DevAuthToken;
+
+			this->ShowTaskEndedNotification(TEXT("Developer Authentication Token Updated"), SNotificationItem::CS_Success);
+		});
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoGenerateDevAuthToken()
+{
+	bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, bIsRunningInChina]
+		{
+			AsyncTask(ENamedThreads::GameThread, [this]()
+				{
+					this->ShowTaskStartedNotification(TEXT("Generating Development Authentication Token"));
+				});
+
+			FString DevAuthToken;
+			if (SpatialCommandUtils::GenerateDevAuthToken(bIsRunningInChina, DevAuthToken))
```

Comment:
we generally prefer to early out instead of having an if-else in the case where one of the cases is a failure. Though here it is short enough that it's probably ok. just something to keep in mind.